src/net.cpp

```diff
@@ -2201,3 +2201,11 @@ bool CNode::SupportsBloomThinBlocks() const {
 bool CNode::SupportsXThinBlocks() const {
     return nServices & NODE_THIN;
 }
+
+bool CNode::SupportsCompactBlocks() const {
+
+    // If a Core node activates segwit, it also
+    // disables support for compact blocks v1
+    return nVersion >= SHORT_IDS_BLOCKS_VERSION
+        && !(nServices & NODE_WITNESS);
```

Comment:
You're right. I was under the impression that the flag would be set after activation. However, it is  set if activation criteria is defined: https://github.com/bitcoin/bitcoin/blob/ed8d693c71b01caa310922f883b3915c09bb6a35/src/init.cpp#L1487

Also, that comment/criteria is based on old Core behavior. I believe they have added support for v1 after segwit activation. We only need to check nVersion now. Good find @rebroad , thanks